manage_containers.py

```diff
@@ -71,16 +71,17 @@ def add(self):
 
     def rm(self):
         """Remove the container if it exists and is not running"""
-        cmd = self.runtime_cmd + ['ps']
-        out = subprocess.run(cmd, text=True, check=True, stdout=subprocess.PIPE).stdout
-        find = rf'({self.clang_tag}|{self.gcc_tag})'
-        running = re.findall(find, out)
-        if not running:
+        out = ''
+        cmd = self.runtime_cmd + ['ps', '-a', '--filter', f'ancestor={self.id}', '--format', '{{.ID}}']
+        container_id = subprocess.run(cmd, text=True, check=True, stdout=subprocess.PIPE).stdout.strip()
```

Comment:
There is a race condition anyway: the container may start just after this check.

How about trying to remove all images without crashing if the container is running?
In this case we can print a message and just proceed to the next image.